build.rs

```diff
@@ -62,23 +75,53 @@ fn main() -> anyhow::Result<()> {
     drop(Command::new("rustfmt").arg(&output).status());
     Ok(())
 }
-
 fn test_directory_module(
     out: &mut String,
     path: impl AsRef<Path>,
     strategy: &str,
 ) -> anyhow::Result<usize> {
     let path = path.as_ref();
     let testsuite = &extract_name(path);
-    with_test_module(out, testsuite, |out| test_directory(out, path, strategy))
+    with_test_module(out, testsuite, |out| {
+        test_directory(out, path, strategy, true, true)
+    })
+}
+
+fn test_directory_module_no_pooling(
+    out: &mut String,
+    path: impl AsRef<Path>,
+    strategy: &str,
+) -> anyhow::Result<usize> {
+    let path = path.as_ref();
+    let testsuite = &extract_name(path);
+    with_test_module(out, testsuite, |out| {
+        test_directory(out, path, strategy, false, true)
+    })
 }
 
+/* FIXME: disabled because of OOM killer
+fn test_directory_module_no_ref_no_pooling(
+    out: &mut String,
+    path: impl AsRef<Path>,
+    strategy: &str,
+) -> anyhow::Result<usize> {
+    let path = path.as_ref();
+    let testsuite = &extract_name(path);
+    with_test_module(out, testsuite, |out| {
+        test_directory(out, path, strategy, false, false)
+    })
+}
+*/
+
 fn test_directory(
     out: &mut String,
     path: impl AsRef<Path>,
     strategy: &str,
+    allow_pooling: bool,
+    reference_types: bool,
 ) -> anyhow::Result<usize> {
     let path = path.as_ref();
+    println!("cargo:rerun-if-changed={}", path.display());
```

Comment:
I do realize that this fixes the issue of "I added a `*.wast` file and would like it to get picked up", but personally I'd prefer to leave this out if you're ok with it. I find it much more useful to work on a `*.wast` file and have a quick cycle time with `cargo test --test all the_test` personally and having this here would force a rebuild of the test binary for any edit to the `*.wast` files.

Reply:
removed